src/v/cloud_storage_clients/abs_client.cc

```diff
@@ -345,6 +351,38 @@ abs_request_creator::make_get_account_info_request() {
     return header;
 }
 
+result<http::client::request_header>
+abs_request_creator::make_set_expiry_to_blob_request(
+  bucket_name const& name,
+  object_key const& key,
+  ss::lowres_clock::duration expires_in) const {
+    // https://learn.microsoft.com/en-us/rest/api/storageservices/set-blob-expiry?tabs=microsoft-entra-id
+    // available only if HNS are enabled for the bucket
+    // performs curl -v -X PUT -H "Authorization: Bearer ${AUTH_CODE}" -H
+    // "x-ms-version: 2023-01-03" -H "x-ms-expiry-option: RelativeToNow" -H
+    // "x-ms-expiry-time: 30000" -d {}
+    // "https://testingimds2ab.blob.core.windows.net/testingcontainer/testHNS?comp=expiry"
+
+    auto header = http::client::request_header{};
+
+    header.method(boost::beast::http::verb::put);
+    header.target(fmt::format("/{}/{}?comp=expiry", name(), key().string()));
```

Comment:
follow-up add unit test to ensure that the target is properly formatted.
an improperly formatted target might result in 400 instead of 404

(an example:
```
key()          -> testingcontainer/"testHNS"?comp=expiry -> 400
key().string() -> testingcontainer/testHNS?comp=expiry   -> 404
```